overlord/devicestate/handlers_install.go

```diff
@@ -691,6 +692,64 @@ func readPreseedAssertion(st *state.State, model *asserts.Model, ubuntuSeedDir,
 
 var seedOpen = seed.Open
 
+type preseedSnapHandler struct {
```

Comment:
could you add a TODO about considering reusing this kind of handler also for UC20 seeding in general